src/useMemoryRouter.test.tsx

```diff
@@ -131,6 +127,27 @@ export function useRouterTests(singletonRouter: MemoryRouter, useRouter: () => R
     });
     expect(result.current.locale).toBe("en");
   });
+
+  it("should preserve the router's reference, but trigger a re-render", async () => {
+    singletonRouter.setCurrentUrl("/page");
+
+    function TestComponent() {
+      const router = useRouter();
+
+      // This should not result in an infinite loop
+      useEffect(() => {
+        router.replace("/page?foo=bar");
+      }, [router]);
```

Comment:
As per my understanding, this actually _should_ cause an infinite loop.  The `router` has "static" methods, like `router.push` and `router.replace`, but also has dynamic properties, like `router.query`.  

You should avoid the infinite loop by doing this instead:
```suggestion
      }, [router.replace]);
```